src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1287,6 +1385,189 @@ std::string MFEMSidreDataCollection::getElementName(mfem::Element::Type elementE
   return "unknown";
 }
 
+mfem::Geometry::Type MFEMSidreDataCollection::getElementTypeFromName(
+  const std::string& name)
+{
+  if(name == "point")
+    return mfem::Geometry::POINT;
+  else if(name == "line")
+    return mfem::Geometry::SEGMENT;
+  else if(name == "tri")
+    return mfem::Geometry::TRIANGLE;
+  else if(name == "quad")
+    return mfem::Geometry::SQUARE;
+  else if(name == "tet")
+    return mfem::Geometry::TETRAHEDRON;
+  else if(name == "hex")
+    return mfem::Geometry::CUBE;
+  else
+    return mfem::Geometry::INVALID;
+}
+
+// private method
+void MFEMSidreDataCollection::reconstructMesh()
+{
+  // mfem::ConduitDataCollection::BlueprintMeshToMesh would be useful here, but
+  // we need all the parameters to construct a ParMesh (the Mesh base subobject
+  // is initialized manually)
+
+  SLIC_ERROR_IF(
+    !verifyMeshBlueprint(),
+    "Cannot reconstruct mesh, data does not satisfy Conduit Blueprint");
+
+  SLIC_ERROR_IF(!m_bp_grp->hasView("coordsets/coords/values/x"),
+                "Cannot reconstruct a mesh without a Cartesian coordinate set");
+
+  View* vertex_view = m_bp_grp->getView("coordsets/coords/values/x");
+  SLIC_ERROR_IF(vertex_view->isExternal(),
+                "Cannot reconstruct a mesh that was built using external data");
+
+  // Assumes that Vertex is a standard layout type with only an array member
+  static_assert(std::is_standard_layout<mfem::Vertex>::value,
```

Comment:
Should we be using SLIC_ASSERT for this?

Reply:
This is a compile-time assertion, my understanding is that SLIC_ASSERTs are for runtime assertions

Reply:
yepp your right carry on